pkg/queue/prometheus_stats_reporter.go

```diff
@@ -75,17 +76,17 @@ type PrometheusStatsReporter struct {
 }
 
 // NewPrometheusStatsReporter creates a reporter that collects and reports queue metrics.
-func NewPrometheusStatsReporter(namespace, config, revision, pod string) (*PrometheusStatsReporter, error) {
-	if namespace == "" {
+func NewPrometheusStatsReporter(env util.Config) (*PrometheusStatsReporter, error) {
+	if env.ServingNamespace == "" {
```

Comment:
Can we remove those checks? Since they are marked as `required` in the config, and as such won't be empty at this point.

Reply:
Let‘s also please not pass the entirety of env into virtually anywhere. I‘d keep the interfaces as they were.